### ISSUE TITLE
checkio sync fails with json decoder error

### ISSUE THREAD

**Issue by nmay231** [CheckiO/checkio-client#19]
### Bug
`checkio sync` fails with `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`. It sounds like it's receiving an empty string from the api.

Apparently, this had worked at some point before because I had a directory with all my solutions. But I wasn't able to sync when I tried again recently.

### Setup
Remove configuration to start from scratch.
```bash
$ rm -rf ~/.checkio # or mv ~/.checkio ~/.checkio_old
$ rm -rf path/to/checkio_solutions # or simply use a new directory
```

Create a virtualenv (I'll be using poetry), re-setup configuration, and try to sync.
```bash
$ poetry init -n
$ poetry add checkio_client@0.2.10
$ poetry run checkio config
# ... Setup config ...
# I selected https://py.checkio.org for the code domain
$ poetry run checkio sync
!!Error!!
```

<details>
<summary>
&lt Full error message &gt
</summary>

```
$ poetry run checkio sync
Using folder "/home/nmay/cur/checkio_test"
Requesting...
Traceback (most recent call last):
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/bin/checkio", line 8, in <module>
    sys.exit(main())
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/lib/python3.8/site-packages/checkio_client/runner.py", line 218, in main
    getattr(module, func_name)(args)
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/lib/python3.8/site-packages/checkio_client/actions/sync.py", line 70, in main
    data = get_user_missions()
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 17, in api_call
    return globals()[func_name + '_' + conf.default_domain_data['game']](*args, **kwargs)
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 97, in get_user_missions_cio
    return api_request('/api/user-missions/')
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 17, in api_call
    return globals()[func_name + '_' + conf.default_domain_data['game']](*args, **kwargs)
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-test-Cbfuw_NV-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 36, in api_request_cio
    resp = json.loads(e.read().decode('utf-8'))
  File "/home/nmay/.asdf/installs/python/3.8.6/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/home/nmay/.asdf/installs/python/3.8.6/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/nmay/.asdf/installs/python/3.8.6/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

</details>

### Versions
- Python 3.8.6
- Checkio client: 0.2.10 (also tried 0.2.9 and 0.2.8)
- OS: Ubuntu 20.04
- Code domain: Python regular Checkio (not Empire of Code)
- Checkio username: "nmay" (in case you need it)

**Comment by oduvan** [comment]
check. Should be fixed now

**Comment by nmay231** [comment]
Unfortunately, version `0.2.12` did not work. It still gives the exact same error. Let me know if/how I can give you more helpful information.

<details>

```
$ poetry run checkio sync # This is version 0.2.12
Using folder "/home/nmay/cur/checkio"
Requesting...
Traceback (most recent call last):
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/bin/checkio", line 8, in <module>
    sys.exit(main())
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages/checkio_client/runner.py", line 224, in main
    getattr(module, func_name)(args)
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages/checkio_client/actions/sync.py", line 70, in main
    data = get_user_missions()
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 17, in api_call
    return globals()[func_name + '_' + conf.default_domain_data['game']](*args, **kwargs)
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 97, in get_user_missions_cio
    return api_request('/api/user-missions/')
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 17, in api_call
    return globals()[func_name + '_' + conf.default_domain_data['game']](*args, **kwargs)
  File "/home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages/checkio_client/api.py", line 36, in api_request_cio
    resp = json.loads(e.read().decode('utf-8'))
  File "/home/nmay/.asdf/installs/python/3.8.6/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/home/nmay/.asdf/installs/python/3.8.6/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/nmay/.asdf/installs/python/3.8.6/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

</details>

As a side note, I wasn't able to install the latest version from pypi. It failed because the source distribution (`*.tar.gz`) didn't include `requirements.txt`. I had to clone the git repo, build a wheel myself, and install from that. Here's the error message from trying to install normally with pip.

<details>

<summary>
$ pip install checkio_client
</summary>

```
Collecting checkio_client
  Using cached checkio_client-0.2.12.tar.gz (33 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ptflk5lg/checkio-client_a17af812bfdd45fab6c52d77424fda1d/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ptflk5lg/checkio-client_a17af812bfdd45fab6c52d77424fda1d/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-d1lyz4hv
         cwd: /tmp/pip-install-ptflk5lg/checkio-client_a17af812bfdd45fab6c52d77424fda1d/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-ptflk5lg/checkio-client_a17af812bfdd45fab6c52d77424fda1d/setup.py", line 5, in <module>
        with open('requirements.txt') as f:
    FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/7b/cc/a1a41b24c3236dffa740fd5d1318d755dbb0982d9749a400cca268f70f48/checkio_client-0.2.12.tar.gz#sha256=fb8db5471646daa21c1fa95b98442175d0e800d6c0c96f1fff7b5cc89d149bed (from https://pypi.org/simple/checkio-client/) (requires-python:>=3.6). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading checkio_client-0.2.11.tar.gz (33 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ptflk5lg/checkio-client_2348fbf796384757949d8ecb28b14942/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ptflk5lg/checkio-client_2348fbf796384757949d8ecb28b14942/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-v8q74298
         cwd: /tmp/pip-install-ptflk5lg/checkio-client_2348fbf796384757949d8ecb28b14942/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-ptflk5lg/checkio-client_2348fbf796384757949d8ecb28b14942/setup.py", line 5, in <module>
        with open('requirements.txt') as f:
    FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/c3/e0/1343385526fe916bc22202944aab306417056ab407a57c7a5fd1b046fb9b/checkio_client-0.2.11.tar.gz#sha256=391892caca9d2e0fc0f74b99da88df7ef50c8e9509e33a214520dfee42b3576c (from https://pypi.org/simple/checkio-client/) (requires-python:>=3.6). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading checkio_client-0.2.10-py3-none-any.whl (52 kB)
     |████████████████████████████████| 52 kB 2.2 MB/s 
Requirement already satisfied: requests==2.7 in /home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages (from checkio_client) (2.7.0)
Requirement already satisfied: websockets==7.0 in /home/nmay/.cache/pypoetry/virtualenvs/checkio-7m5IZAsx-py3.8/lib/python3.8/site-packages (from checkio_client) (7.0)
Installing collected packages: checkio-client
Successfully installed checkio-client-0.2.10
```

</details>

**Comment by nmay231** [comment]
@oduvan

**Comment by nmay231** [comment]
I did a little digging myself and saw that the root issue is requesting `/api/user-missions` gets a 502 HTTP error.

And the issue where I could not install the client from pypi has been resolved since #21.
